### PR TITLE
Feat clean command

### PR DESCRIPTION
**Pull request by sleepyqadir**
closes https://github.com/onlydustxyz/cairo-foundry/issues/68

this PR will create a clean command for cleaning the cache and compiled file for each contract in os_cache_dir/cairo-foundry-cache and os_cache_dir/compiled-cairo-files where os_cache_dir is different for each os

### CODE REVIEW

File: src/cli/commands/clean/mod.rs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	if dir.exists() {
+		fs::remove_dir_all(dir).map_err(|err| err.to_string())?;
+	}
+
+	Ok(())
+}
+
+impl CommandExecution<CleanOutput> for CleanArgs {
+	fn exec(&self) -> Result<CleanOutput, String> {
+		clear_directory("cairo-foundry-cache").unwrap();
```
Comment (Eikix) [inline src/cli/commands/clean/mod.rs:42]: could we turn these hardcoded strings to globally available const strings (screaming case) in the project? that way if one day they change they change everywhere at once

File: src/cli/commands/clean/mod.rs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	let path_to_cache_dir = dirs::cache_dir()
+		.ok_or(Error::CacheDirSupported)
+		.map_err(|err| err.to_string())?;
+
+	let mut dir = PathBuf::new();
+	dir.push(&path_to_cache_dir);
+	dir.push(path);
+
+	if dir.exists() {
+		fs::remove_dir_all(dir).map_err(|err| err.to_string())?;
```
Comment (Eikix) [inline src/cli/commands/clean/mod.rs:34]: could you use the new error management system here?

File: src/cli/commands/mod.rs [hunk 3]
```diff
@@ -55,7 +62,9 @@ impl CommandExecution<Output> for Commands {
 	fn exec(&self) -> Result<Output, String> {
 		match &self {
 			Commands::List(args) => args.exec().map(|o| Output(CommandOutputs::List(o))),
+			Commands::Execute(args) => args.exec().map(|o| Output(CommandOutputs::Execute(o))),
```
Comment (Eikix) [inline src/cli/commands/mod.rs:65]: Why do you need to map Execute?

**Comment by Eikix** [comment]
Btw are you on the Cairo-foundry discord group?

**Comment by tdelabro** [comment]
@sleepyqadir are you still working on this?

**Comment by tdelabro** [comment]
replaced by #116

**Pull request closed without merging**